tracer/test/Datadog.Trace.Security.Unit.Tests/ObjectExtractorTests.cs

```diff
@@ -121,6 +122,21 @@ public void TestStruct()
             Assert.Equal(target.StringValue, result[nameof(target.StringValue)]?.ToString());
         }
 
+        [Fact]
+        public void TestAnonymousType()
+        {
+            var target = new { Dog1 = "test", Dog2 = "test2", Id = 1 };
+
+            var result = ObjectExtractor.Extract(target) as Dictionary<string, object>;
+
+            Assert.NotNull(result);
+
+            Assert.Equal(target.Dog1, result[nameof(target.Dog1)]?.ToString());
+            Assert.Equal(target.Dog2, result[nameof(target.Dog2)]?.ToString());
+            result[nameof(target.Id)].Should().BeOfType<int>();
+            target.Id.Should().Be((int)result[nameof(target.Id)]);
+        }
```

Comment:
Can I suggest a few more tests to make sure it works as you'd like:

```csharp
var target = new {}; // empty
var target = new { Name = "Outer", Dog = new { Name = "Inner" }, Cat = new {} }; // nested with same properties + empty
var target = new [] { new{ Name = "Anon1" }, new{ Name = "Anon2" }, new{ Name = "Anon1" } ]; // array of anon
```

Reply:
thank you added in [2725680](https://github.com/DataDog/dd-trace-dotnet/pull/4865/commits/2725680c77513375b1d3b496c8f6e9cb4cfe3b23)